Assets/Scripts/Game Controllers/LevelEndTriggerController.cs

```diff
@@ -10,13 +10,22 @@ public class LevelEndTriggerController : MonoBehaviour
 
     private void Start()
     {
-        indexForNextScene = SceneManager.GetActiveScene().buildIndex + 1;
+        if (SceneManager.GetActiveScene().buildIndex == 0)
+        {
+            indexForNextScene = lastLevel + 1;
+        }
+        else
+        {
+            indexForNextScene = SceneManager.GetActiveScene().buildIndex + 1;
+        }
     }
 
     private void OnTriggerEnter2D(Collider2D collision)
     {
         if (collision.CompareTag("Player"))
         {
+            ScoreManager.Instance.LevelEnd(collision.transform.position.y);
```

Comment:
I think it's better for `ScoreManager` to have a reference to the player's transform. Then, `ScoreManager` can get the start and end Y positions of the player rather than needing a reference to the player's transform in order to end the level properly (even though I don't think it's likely the level will be ended in a script that doesn't have a reference to the player).